app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -507,6 +507,10 @@ public function saveEntity($entity, $unlock = true)
             $details = $ips->first()->getIpDetails();
             // Only update with IP details if none of the following are set to prevent wrong combinations
             if (empty($fields['core']['city']['value']) && empty($fields['core']['state']['value']) && empty($fields['core']['country']['value']) && empty($fields['core']['zipcode']['value'])) {
+                if ($this->coreParametersHelper->getParameter('anonymize_ip') && $this->ipLookupHelper->getRealIp()) {
+                    $details = $this->ipLookupHelper->getIpDetails($this->ipLookupHelper->getRealIp());
```

Comment:
On this call if `$this->ipLookup == NULL` within `getIpDetails` function an error is thrown.
`Call to a member function setIpAddress() on null`

Reply:
With last commit https://github.com/mautic/mautic/pull/8175/commits/aa3139c8c9792aed335fbc1e176a8f007635a2aa, `$details from line 507` is erased by an empty array `if $this->ipLookup == NULL`, is this wanted behavior?

Reply:
If ipLookup is null then there is fatal error. Do you have any experience?
But I noticed another warning, I will check it today

https://github.com/mautic/mautic/pull/8175/files#diff-680a217018ae349738fa38e34b7a19f8R228

Reply:
But there is not reason $details contain something. 
If we use anonymize_ip then we don't have store any details if IP address (127.0.0.*** etc).
This reason why we ask details of real IP in this step

Reply:
Ok, perfect.